File: exercises/solution1.py
import sys

#tratando input do programa
if len(sys.argv) < 3:
    print(f"Uso: {sys.argv[0]} valor1 valor2 ... valorN valor_referencia")
    sys.exit(1)

#carregando os vetores
valores = list(map(float, sys.argv[1:-1]))

#lendo o valor de referencia
referencia = float(sys.argv[-1])

#calculando a media
media = sum(valores) / len(valores)

# gerando saida do programa
print(f"Média: {media}")
print(f"Valor de referência: {referencia}")
if media > referencia:
    print("A média é maior que o valor de referência.")
else:
    print("A média NÃO é maior que o valor de referência.")

File: exercises/solution1.cpp
#include <iostream>
#include <vector>
#include <cstdlib> // para atof
using namespace std;

int main(int argc, char* argv[]) {
    if (argc < 3) {
        cerr << "Uso: " << argv[0] << " valor1 valor2 ... valorN valor_referencia" << endl;
        return 1;
    }

    vector<double> valores;
    for (int i = 1; i < argc - 1; i++) {
        valores.push_back(atof(argv[i]));
    }

    double referencia = atof(argv[argc - 1]);

    double soma = 0;
    for (double v : valores) {
        soma += v;
    }
    double media = soma / valores.size();

    cout << "Média: " << media << endl;
    cout << "Valor de referência: " << referencia << endl;
    cout << (media > referencia ? "A média é maior que o valor de referência." 
                                : "A média NÃO é maior que o valor de referência.") 
         << endl;

    return 0;
}